src/main/java/net/minecraftforge/items/wrapper/PlayerInvWrapper.java

```diff
@@ -0,0 +1,69 @@
+package net.minecraftforge.items.wrapper;
+
+import net.minecraft.entity.player.InventoryPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.items.IItemHandlerModifiable;
+
+public class PlayerInvWrapper implements IItemHandlerModifiable
+{
```

Comment:
Wouldn't this just be better a a 'chained' inventory wrapper?
have it all driven by a 'getInventory(slot)' function and if (inv == null) invalid slot! ?
Would be more generic then player spefic things.